### ISSUE TITLE
[bug] App window when developing gets hidden behind all other windows

### ISSUE THREAD

**Issue by darricheng** [tauri-apps/tauri#8023]
### Describe the bug

After I run `cargo tauri dev`, the app GUI should launch automatically and immediately float above all other windows. That was the behaviour before. I would be able to switch to the app using the Mac OS app switcher too.

I think this happened after I updated my OS to Sonoma (i.e. Mac OS 13 to 14). Currently, after running `cargo tauri dev`, the GUI launches, but it seems to no longer be registered by the window manager of Mac OS. The open app no longer appears in the app switcher. When I switch to the app window using Mission Control (shortcut is swiping up with 3 fingers on the keyboard, or ctrl+up), the window immediately disappears behind all my other windows.

### Reproduction

1. Follow the instructions on the [tauri website](https://tauri.app/v1/guides/getting-started/setup/sveltekit/) to set up a new sveltekit project.
2. Run `cargo tauri dev` to start the dev environment.

### Expected behavior

The app GUI should be able to appear on top of other open windows.

### Platform and versions

```text
[‚úî] Environment
    - OS: Mac OS 14.0.0 X64
    ‚úî Xcode Command Line Tools: installed
    ‚úî rustc: 1.71.1 (eb26296b5 2023-08-03)
    ‚úî Cargo: 1.71.1 (7f1d04c00 2023-07-29)
    ‚úî rustup: 1.26.0 (5af9b9484 2023-04-05)
    ‚úî Rust toolchain: stable-aarch64-apple-darwin (environment override by RUSTUP_TOOLCHAIN)
    - node: 20.5.1
    - pnpm: 8.6.12
    - yarn: 1.22.19
    - npm: 9.8.0

[-] Packages
    - tauri [RUST]: 1.5.2
    - tauri-build [RUST]: 1.5.0
    - wry [RUST]: 0.24.4
    - tao [RUST]: 0.16.4
    - @tauri-apps/api [NPM]: not installed!
    - @tauri-apps/cli [NPM]: 1.4.0 (outdated, latest: 1.5.2)

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../build
    - devPath: http://localhost:5173/
    - framework: Svelte
    - bundler: Vite
```

### Stack trace

```text
-
```

### Additional context

The major change is upgrading my Mac OS from 13 to 14.

**Comment by darricheng** [comment]
Additional information: I am not able to focus the application, i.e. the active application shown at the top right of the MacOS menu bar doesn't change to the tauri app being developed. This inability to focus results in me not being able to input text in the app while developing, such that I'm not able to properly test features.

I appreciate any advice to further troubleshoot this issue. 🙏🏻

**Comment by darricheng** [comment]
Okay so I'm no longer facing this issue. 

Since the last comment, what I did was the following:

1. Tried to run `cargo tauri build`, unable to due to having the default bundle identifier `com.tauri.dev`.
2. Update the identifier, then build and test the built app.
3. Retry `cargo tauri dev`, app is able to appear above other apps and I'm able to input text into the app. The active application shown at the top right of the MacOS menu bar also correctly shows the app I'm developing.
4. Revert the identifier, try `cargo tauri dev` again, and like step 3 the dev app works as expected.

Unfortunately, I'm not able to further test this issue as I didn't actually try `cargo tauri dev` right before `cargo tauri build`.

My guess is that perhaps MacOS didn't register the app prior to building the app once, but after building it was able to properly register the app to its window manager. But frankly, I have no clue.

I'll close this issue as I no longer seem to face this.
